cmake/Platform/Android/CMakePresets.json

```diff
@@ -0,0 +1,256 @@
+        "unity"
+      ],
+      "condition": {
+        "type": "inList",
+        "string": "${hostSystemName}",
+        "list": [
+          "Darwin",
+          "Linux",
+          "Windows"
+        ]
+      }
+    },
+    {
+      "name": "android-no-unity",
+      "displayName": "Android without Unity",
+      "description": "Android build which uses unity files",
+      "toolchainFile": "${sourceDir}/cmake/Platform/Android/Toolchain_android.cmake",
+      "binaryDir": "${sourceDir}/build/android_no_unity",
+      "inherits": [
+        "no-unity"
+      ],
+      "condition": {
+        "type": "inList",
+        "string": "${hostSystemName}",
+        "list": [
+          "Darwin",
+          "Linux",
+          "Windows"
+        ]
+      }
+    },
+    {
+      "name": "android-non-monolithic",
+      "displayName": "Android Non-Monolithic",
+      "description": "Default configuration for non-monolithic builds",
+      "toolchainFile": "${sourceDir}/cmake/Platform/Android/Toolchain_android.cmake",
+      "inherits": [
+        "non-monolithic"
+      ],
+      "condition": {
+        "type": "inList",
+        "string": "${hostSystemName}",
+        "list": [
+          "Darwin",
+          "Linux",
+          "Windows"
+        ]
+      }
+    },
+    {
+      "name": "android-monolithic",
+      "displayName": "Android Monolithic",
+      "description": "Default configuration for monolithic builds",
+      "toolchainFile": "${sourceDir}/cmake/Platform/Android/Toolchain_android.cmake",
+      "binaryDir": "${sourceDir}/build/android_mono",
+      "inherits": [
+        "monolithic"
+      ],
+      "condition": {
+        "type": "inList",
+        "string": "${hostSystemName}",
+        "list": [
+          "Darwin",
+          "Linux",
+          "Windows"
+        ]
+      }
+    },
+    {
+      "name": "android-ninja",
+      "displayName": "Android Ninja",
+      "description": "Configure Android using with the Ninja generator",
+      "toolchainFile": "${sourceDir}/cmake/Platform/Android/Toolchain_android.cmake",
+      "binaryDir": "${sourceDir}/build/android_ninja",
+      "inherits": [
+        "android-ninja-unity"
+      ],
+      "condition": {
+        "type": "inList",
+        "string": "${hostSystemName}",
+        "list": [
+          "Darwin",
+          "Linux",
+          "Windows"
+        ]
+      }
+    },
+    {
+      "name": "android-ninja-unity",
+      "displayName": "Android Ninja Unity",
+      "description": "Configure Android with the Ninja generator + Unity Builds",
+      "toolchainFile": "${sourceDir}/cmake/Platform/Android/Toolchain_android.cmake",
+      "binaryDir": "${sourceDir}/build/android_ninja_unity",
+      "inherits": [
+        "ninja-multi-config",
+        "android-unity"
+      ],
+      "condition": {
+        "type": "inList",
+        "string": "${hostSystemName}",
+        "list": [
+          "Darwin",
+          "Linux",
+          "Windows"
+        ]
+      }
```

Comment:
Is this condition really necessary? When is the host system name not one of Darwin, Linux, or Windows?

Reply:
For O3DE purposes, it is never not one on these platforms, but this is here in case we want to remove support for building Android on a platform.
I know we have Android is able to build on Windows and we have built it on MacOS in the past, but on Linux we haven't actually had someone built it there, but it should work :tm:?

In case it doesn't work we can just remove the "Linux" entry from the list.